src/app/components/BibPage/Tabbed.jsx

```diff
@@ -17,105 +20,127 @@ class Tabbed extends React.Component {
-    if (this.props.hash) {
-      let hash = this.props.hash;
-      const hashMatch = this.props.hash.match(/[^\d]*(\d)/);
+    const { hash } = this.props;
+    if (hash) {
+      const hashMatch = hash.match(/[^\d]*(\d)/);
       if (hashMatch) hashNumber = hashMatch[1];
       window.location.replace(window.location.href.replace(/#.*/, '') + hash);
-      let tab = this.links[hashNumber];
+      const tab = this.links[hashNumber];
       tab.focus();
     }
     this.setState({ tabNumber: hashNumber.toString() });
   }
 
   focusTab(newTabIndex) {
-    let newTab = this.links[newTabIndex];
+    const newTab = this.links[newTabIndex];
     newTab.focus();
   }
 
-  //switches tabs by updating state and href
+  // switches tabs by updating state and href
   switchTab(newTabIndex) {
     if (newTabIndex !== this.state.tabNumber) {
-      const tabChoices = ['Details Tab1', 'Full Description Tab2'];
+      const tabChoices = ['Availibility Tab1', 'Details Tab2', 'Full Description Tab3'];
       trackDiscovery('BibPage Tabs Switch', tabChoices[newTabIndex - 1]);
     }
     this.setState({ tabNumber: newTabIndex.toString() });
-    let newTab = this.links[newTabIndex];
-    window.location.replace(window.location.href.split('#')[0] + `#tab${newTabIndex}`);
+    const newTab = this.links[newTabIndex];
+    window.location.replace(`${window.location.href.split('#')[0]}#tab${newTabIndex}`);
     newTab.focus();
   }
 
   clickHandler(e) {
     e.preventDefault();
-    let clickedTab = e.currentTarget;
-    let index = clickedTab.getAttribute('data');
+    const clickedTab = e.currentTarget;
+    const index = clickedTab.getAttribute('data');
     this.switchTab(index);
   }
 
-  //enables navigation with arrow keys
+  // enables navigation with arrow keys
   keyDownHandler(e) {
-    let panel = window.location.href.split("#")[1] ? this.sections[this.state.tabNumber] : this.default;
-    const index = parseInt(e.currentTarget.getAttribute('data'));
-    let dir = e.which === 37 ? index - 1 : e.which === 39 ? index + 1 : e.which === 40 ? 'down' : null;
+    const panel = window.location.href.split('#')[1] ? this.sections[this.state.tabNumber] : this.default;
+    const index = parseInt(e.currentTarget.getAttribute('data'), 10);
+    const getDir = () => {
+      switch (e.which) {
+        case 37:
+          return index - 1;
+        case 39:
+          return index + 1;
+        case 40:
+          return 'down';
+        default:
+          return null;
+      }
+    };
+    const dir = getDir();
     if (e.which === 32) {
       e.preventDefault();
       this.clickHandler(e);
     }
     if (dir !== null) {
       e.preventDefault();
-      dir === 'down' ? panel.focus() : dir <= this.state.numberOfTabs && 0 <= dir ? this.focusTab(dir) : void 0;
+      if (dir === 'down') panel.focus();
+      else if (dir <= this.state.numberOfTabs && dir >= 0) this.focusTab(dir);
     }
   }
 
 
   render() {
+    const getTabIndex = (j) => {
```

Comment:
Factoring this out is a good idea but why not make it a method in this class instead of defining it in the render method?

Reply:
I see it more as a small helper method for building the render, rather than related to the behavior of the component.